File: Bai_21/main.py
# # Đọc dữ liệu từ file input
with open('E:\Language\Baitaptonghop\Bai_21\HANGXOM.INP','r') as file:
    L, R = map(int, file.read().split())

# Function to check if a number is prime
def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

# Function to count prime pairs with a difference of 2 in a range
def count_prime_pairs(L, R):
    count = 0
    for i in range(L, R):
        if is_prime(i) and is_prime(i + 2):
            count += 1
    return count

# Counting prime pairs and printing the result
result = count_prime_pairs(L, R + 1)  # Adding 1 to R to include R in the range

# Ghi kết quả ra file output
with open('E:\Language\Baitaptonghop\Bai_21\HANGXOM.OUT', 'w') as file:
    file.write(str(result))

File: Bai_21/main.cpp
#include <iostream>
#include <fstream>
#include <cmath>

using namespace std;
// Function to check if a number is prime
bool is_prime(int num)
{
    if (num < 2)
    {
        return false;
    }
    for (int i = 2; i <= sqrt(num); ++i)
    {
        if (num % i == 0)
        {
            return false;
        }
    }
    return true;
}

// Function to count prime pairs with a difference of 2 in a range
int count_prime_pairs(int L, int R)
{
    int count = 0;
    for (int i = L; i < R; ++i)
    {
        if (is_prime(i) && is_prime(i + 2))
        {
            count++;
        }
    }
    return count;
}

int main()
{
    ifstream infile("HANGXOM.INP");
    ofstream outfile("HANGXOM.OUT");
    int L, R;
    infile >> L >> R;

    // Counting prime pairs and printing the result
    int result = count_prime_pairs(L, R + 1); // Adding 1 to R to include R in the range
    outfile << result;

    return 0;
}
